core/src/avm2/api_version.rs

```diff
@@ -0,0 +1,60 @@
+// Based on https://github.com/adobe/avmplus/blob/master/core/api-versions.h
+// FIXME - if we ever add in AIR emulation, then implement the 'version series' logic.
+// For now, it's fine to just compare Flash player version against air versions directly.
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, FromPrimitive, Debug)]
+#[allow(non_camel_case_types)]
+pub enum ApiVersion {
+    AllVersions = 0,
+    AIR_1_0 = 1,
+    FP_10_0 = 2,
+    AIR_1_5 = 3,
+    AIR_1_5_1 = 4,
+    FP_10_0_32 = 5,
+    AIR_1_5_2 = 6,
+    FP_10_1 = 7,
+    AIR_2_0 = 8,
+    AIR_2_5 = 9,
+    FP_10_2 = 10,
+    AIR_2_6 = 11,
+    SWF_12 = 12,
+    AIR_2_7 = 13,
+    SWF_13 = 14,
+    AIR_3_0 = 15,
+    SWF_14 = 16,
+    AIR_3_1 = 17,
+    SWF_15 = 18,
+    AIR_3_2 = 19,
+    SWF_16 = 20,
+    AIR_3_3 = 21,
+    SWF_17 = 22,
+    AIR_3_4 = 23,
+    SWF_18 = 24,
+    AIR_3_5 = 25,
+    SWF_19 = 26,
+    AIR_3_6 = 27,
+    SWF_20 = 28,
+    AIR_3_7 = 29,
+    SWF_21 = 30,
+    AIR_3_8 = 31,
+    SWF_22 = 32,
+    AIR_3_9 = 33,
+    SWF_23 = 34,
+    AIR_4_0 = 35,
+    SWF_24 = 36,
+    AIR_13_0 = 37,
+    SWF_25 = 38,
+    AIR_14_0 = 39,
+    SWF_26 = 40,
+    AIR_15_0 = 41,
+    SWF_27 = 42,
+    AIR_16_0 = 43,
+    SWF_28 = 44,
+    AIR_17_0 = 45,
+    SWF_29 = 46,
+    AIR_18_0 = 47,
+    SWF_30 = 48,
+    AIR_19_0 = 49,
+    SWF_31 = 50,
+    AIR_20_0 = 51,
```

Comment:
Re-stating what I said on Discord, this is probably not the highest ApiVersion we'll ever need to gate. https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/media/AudioDeviceManager.html was added in Flash Player 27, which according to https://github.com/ruffle-rs/ruffle/wiki/SWF-version-chart, is SWF version 38.

Reply:
FP doesn't actually implement anything higher than that- see [the avmplus header that this file references](https://github.com/adobe/avmplus/blob/master/core/api-versions.h). I'm not sure that FP can actually gate anything higher than that- it might be faking it. It's possible that it uses 
SWF version to version-gate `AudioDeviceManager` and doesn't version-gate it based on AIR version.